sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditionsTest.java

```diff
@@ -272,57 +333,318 @@ void testAssociateNewLocationForHandle() {
+        final Thread t = new Thread(() -> {
+            ManipulationPreconditions.AssociateLocationsManipulation.EpisodicContextReportStateHandleObserver
+                    mdibAccessObserver;
+            for (int i = 0; i < 2; i++) {
+                mdibAccessObserver =
+                        (ManipulationPreconditions.AssociateLocationsManipulation
+                                        .EpisodicContextReportStateHandleObserver)
+                                observer.get();
+                synchronized (signal) {
+                    while (mdibAccessObserver == null) {
+                        try {
+                            signal.wait(300);
+                        } catch (InterruptedException e) {
+                            // do nothing
+                        }
+                        mdibAccessObserver =
+                                (ManipulationPreconditions.AssociateLocationsManipulation
+                                                .EpisodicContextReportStateHandleObserver)
+                                        observer.get();
+                    }
+                }
+                observers.add(mdibAccessObserver);
+                final ContextStateModificationMessage report = new ContextStateModificationMessage(
+                        mockMdibAccess, Map.of("foo", List.of(mockLocationContextState, mockLocationContextState2)));
+
+                // the delay
+                sleep(2000);
+                updated = true;
+                mdibAccessObserver.onUpdate(report);
+
+                observer.set(null);
+            }
+        });
+        t.start();
+
+        assertTrue(
+                ManipulationPreconditions.AssociateLocationsManipulation.manipulation(injector),
+                "Manipulation should've succeeded");
+        assertFalse(
+                testRunObserver.isInvalid(),
+                "Test run should not have been invalid. Reason(s): " + testRunObserver.getReasons());
+
+        final var handleCaptor = ArgumentCaptor.forClass(String.class);
+        final var assocCaptor = ArgumentCaptor.forClass(ContextAssociation.class);
+        verify(mockManipulations, times(expectedManipulationCalls))
+                .createContextStateWithAssociation(handleCaptor.capture(), assocCaptor.capture());
+
+        for (MdibAccessObserver obs : observers) {
+            verify(mockMdibAccessObservable).registerObserver(obs);
+        }
+
+        assertEquals(LOCATION_CONTEXT_DESCRIPTOR_HANDLE, handleCaptor.getValue());
+        assertEquals(ContextAssociation.ASSOC, assocCaptor.getValue());
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private void sleep(final long delayInMillis) {
+        final long start = System.currentTimeMillis();
+        final long end = start + delayInMillis;
+        long now = System.currentTimeMillis();
+        while (now < end) {
+            try {
+                Thread.sleep(end - now);
+            } catch (InterruptedException e) {
```

Comment:
do not catch it here if you are not reacting to it

also you are not even interrupting the thread from another thread, thus there is no good way of reacting to it and it should just be escalated

Reply:
I wrote this method to get a sleep that does NOT throw InterruptedExceptions. When I do not catch them at this point, but declare them as thrown by the method, then I could just use the normal Thread.sleep() method.